src/main/java/com/google/devtools/build/lib/actions/ResourceManager.java

```diff
@@ -410,9 +453,21 @@ private boolean areResourcesAvailable(ResourceSet resources) {
     // 3) If used resource amount is less than total available resource amount.
     boolean cpuIsAvailable = cpu == 0.0 || usedCpu == 0.0 || usedCpu + cpu <= availableCpu;
     boolean ramIsAvailable = ram == 0.0 || usedRam == 0.0 || ram <= remainingRam;
+    boolean extraResourcesIsAvailable = true;
+    for (Map.Entry<String, Float> resource : resources.getExtraResourceUsage().entrySet()) {
+      String key = (String)resource.getKey();
+      float used = (float)usedExtraResources.getOrDefault(key, 0f);
+      float requested = resource.getValue();
+      float available = (float)availableResources.getExtraResourceUsage().getOrDefault(key, 0f);
+      float epsilon = 0.0001f;  // Account for possible rounding errors.
+      if (requested != 0.0 && used != 0.0 && requested + used > available + epsilon) {
+        extraResourcesIsAvailable = false;
+        break;
+      }
+    }
```

Comment:
Maybe it's better to move this code in function to make code more readable.

Reply:
Done.